src/methods/wallet-get-calls-status.test.ts

```diff
@@ -0,0 +1,130 @@
+import type { JsonRpcRequest, PendingJsonRpcResponse } from '@metamask/utils';
+import { klona } from 'klona';
+
+import type {
+  GetCallsStatusParams,
+  GetCallsStatusResult,
+  GetTransactionReceiptsByBatchIdHook,
+} from './wallet-get-calls-status';
+import { walletGetCallsStatus } from './wallet-get-calls-status';
+
+const ID_MOCK = '1234-5678';
+
+const RECEIPT_MOCK = {
+  logs: [
+    {
+      address: '0x123abc123abc123abc123abc123abc123abc123a',
+      data: '0x123abc',
+      topics: ['0x123abc'],
+    },
+  ],
+  status: '0x1',
+  chainId: '0x1',
+  blockHash: '0x123abc',
+  blockNumber: '0x1',
+  gasUsed: '0x1',
+  transactionHash: '0x123abc',
+};
+
+const REQUEST_MOCK = {
+  params: [ID_MOCK],
+} as unknown as JsonRpcRequest<GetCallsStatusParams>;
+
+describe('wallet_getCallsStatus', () => {
+  let request: JsonRpcRequest<GetCallsStatusParams>;
+  let params: GetCallsStatusParams;
+  let response: PendingJsonRpcResponse<GetCallsStatusResult>;
+  let getTransactionReceiptsByBatchIdMock: jest.MockedFunction<GetTransactionReceiptsByBatchIdHook>;
+
+  async function callMethod() {
+    return walletGetCallsStatus(request, response, {
+      getTransactionReceiptsByBatchId: getTransactionReceiptsByBatchIdMock,
+    });
+  }
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    request = klona(REQUEST_MOCK);
+    params = request.params as GetCallsStatusParams;
+    response = {} as PendingJsonRpcResponse<GetCallsStatusResult>;
+
+    getTransactionReceiptsByBatchIdMock = jest
+      .fn()
+      .mockResolvedValue([RECEIPT_MOCK, RECEIPT_MOCK]);
+  });
+
+  it('calls hook', async () => {
+    await callMethod();
+    expect(getTransactionReceiptsByBatchIdMock).toHaveBeenCalledWith(
+      params[0],
+      request,
+    );
+  });
+
+  it('returns confirmed status if all receipts available', async () => {
+    await callMethod();
+    expect(response.result?.status).toBe('CONFIRMED');
+  });
+
+  it('returns pending status if missing receipts', async () => {
```

Comment:
should we also test that `receipts` `null` if there are pending?

Reply:
Done.